display_list/skia/dl_sk_conversions.cc

```diff
@@ -97,7 +103,7 @@ sk_sp<SkShader> ToSk(const DlColorSource* source) {
       SkPoint pts[] = {linear_source->start_point(),
                        linear_source->end_point()};
       return SkGradientShader::MakeLinear(
-          pts, ToSkColors(linear_source), linear_source->stops(),
+          pts, ToSkColors(linear_source).data(), linear_source->stops(),
```

Comment:
Can calling `'data()` inline here potentially allow premature delete of the object?

Reply:
^

Reply:
Nope, rvalues in argument lists are deallocated after the function is executed.

Reply:
Wait, I think I may be wrong about this.  I'll change it.

Reply:
What I said was right, but it didn't apply to what was happening here because of the chaining.  I've fixed it.

Reply:
I was too trusting of the tests and the linter.  That's the sort of thing I would like our linter to flag =T

Reply:
I was even running my tests with guardmalloc and it didn't get flagged at runtime.  This might just happen to work in our setup.